enhancements/console/dynamic-plugins.md

```diff
@@ -149,16 +149,27 @@ spec:
 
 In case the plugin needs to communicate with some in-cluster service, it can
 declare a service proxy in its `ConsolePlugin` resource using the
-`spec.proxy.services` array field. A service `name`, `namespace` and `port`
-needs to be specified.
+`spec.proxy.services` map. The key of the map is the proxy alias
+of the service and value is an object in which service `name`, `namespace` and `port`
+needs to be specified and to which the request will be proxied.
+
+```yaml
+spec:
+  proxy:
+    services:
```

Comment:
I wonder if we need `services` here now. If we want to add other types of proxies, we want the alias to be unique. I'd suggest:

```yaml
spec:
  proxy:
    my-alias:
      service:
         name: my-name
         namespace: my-namespace
         port: my-port
      caCertificate: ...
      authorize: true
```

Then later on we could do something like

```yaml
spec:
  proxy:
    my-alias:
       url: https://example.com/
       caCertificate: ...
```

Reply:
So after talking to @stlaz on the API design here we agreed that best would be to have the `spec.proxy` structured as an array, which would contain `alias`, `authorize`, `caCertificate` and  enum `type` field. 
Based on the `type` a proxy config struct would be used.

eg.
```go
type ConsolePluginSpec struct {
	DisplayName string `json:"displayName"`
	Service ConsolePluginService `json:"service"`
	Proxy []ConsolePluginProxy `json:"proxy"`
}


type ConsolePluginProxy struct {
	Type ProxyType `json:"type"`
	Service ProxyServiceConfig `json:"service"`
	URL ProxyTypeURLConfig `json:"url"`
	CACertificate string `json:"caCertificate"`
	Authorize bool `json:"authorize"`
}

type ProxyType string

const (
	ProxyTypeService ProxyType = "Service"
	ProxyTypeURL ProxyType = "URL"
)

type ProxyTypeServiceConfig struct {
	Name string `json:"name"`
	Namespace string `json:"namespace"`
	Port int32 `json:"port"`
}

type ProxyTypeURLConfig struct {
	URL string `json:"url"`
}
```

This is the idiomatic way how the API should be structured... check[ indentityProviders API for example
](https://github.com/openshift/api/blob/9b3afd95c93af476bec246b2c60f64235f078198/config/v1/types_oauth.go#L164-L239)

Reply:
@jhadvig That looks good to me.